doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+
+A protocol reference implementation is provided with an Astarte SDK, however developers might implement it from scratch using 3rd party libraries with their favourite languages: all formats and protocols described here are open and well documented. Last but not least Astarte doesn't mandate this protocol, and a different one can be used with a different transport.
+
+## MQTT Topics Overview
+
+Astarte MQTT v1 Protocol relies on few well known topics that are reserved.
+
+| Topic                                                  | Purpose          | Published By | QoS     | Payload Format                          |
+|--------------------------------------------------------|------------------|--------------|---------|-----------------------------------------|
+| `<realm name>/<device id>`                             | Introspection    | Device       | 2       | ASCII plain text, ':' and ';' delimited |
+| `<realm name>/<device id>/control/emptyCache`          | Empty Cache      | Device       | 2       | ASCII plain text (always "1")           |
+| `<realm name>/<device id>/control/consumer/properties` | Purge Properties | Astarte      | 2       | deflated plain text                     |
+| `<realm name>/<device id>/control/producer/properties` | Purge Properties | Device       | 2       | deflated plain text                     |
+| `<realm name>/<device id>/<interface name>/<path>`     | Publish Data     | Both         | 0, 1, 2 | BSON (or empty)                         |
+
+For clarity reasons all `<realm name>/<device id>` prefixes will be omitted on the following paragraphs, those topics will be called device topics.
+Topics are not bidirectional, devices must not publish data for server owned topics and viceversa, onwership is explicitly stated in interfaces files.
+
+## BSON
+
+BSON format allows to save precious bytes compared to JSON, while it offers the advantages of a schema-less protocol.
+Previous statement can be explained with two really simple examples, which are the same payload encoded with JSON and BSON:
+
+```
+{"v":25.367812,"t":1537346756844}
+```
+
+counts 33 bytes.
+
+The following hex dump is the same message but encoded with BSON:
+```
+0000000 1b 00 00 00 12 74 00 ec e0 01 f1 65 01 00 00 01
+0000020 76 00 8c 13 5f ed 28 5e 39 40 00
+```
+
+that fits just in 27 bytes.
+
+### BSON format
+
+BSON format is a really simple binary format, so simple that the previous example can be easily broken out: first 4 bytes  (`1b 00 00 00`) are the document size header, follows the timestamp marker (`09`), the timestamp key name (`74 00`, that is "t"), the timestamp value (`5f 48 06 f1 65 01 00 00` as int64), the double value marker (`01`),  the value key name (`76 00`, that is "v"), the actual value (`cd cc cc cc cc 4c 39 40` as 64-bit IEEE 754-2008 floating point) and the end of document marker (`00`).
```

Comment:
* BSON, not BSON format.
* Make first sentence less verbose: Breaking down the previous example is very easy thanks to BSON's simplicity:
* the first 4 bytes
* Watch out for extra spaces between bytes  (`1b 00 00 00`)